app/components/modal-overlay.tsx

```diff
@@ -1,39 +1,48 @@
 "use client";
-import { Suspense } from "react";
-import { useSearchParams } from "next/navigation";
-import Link from "next/link";
+import { useEffect, MouseEvent, ReactNode } from 'react';
 
-function ModalOverlayContent() {
-    const searchParams = useSearchParams();
-    const modal = searchParams.get("modal");
+interface ModalOverlayProps {
+  onClose: () => void;
+  children: ReactNode;
+}
+
+function ModalOverlay({ onClose, children }: ModalOverlayProps) {
+  // Handle outside click to close modal
+  const handleOutsideClick = (e: MouseEvent<HTMLDivElement>) => {
+    if ((e.target as HTMLElement).id === 'modal-overlay') {
+      onClose();
+    }
+  };
+
+  // Close modal on 'Escape' key press
+  useEffect(() => {
+    const handleEscape = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        onClose();
+      }
+    };
+    window.addEventListener('keydown', handleEscape);
+    return () => window.removeEventListener('keydown', handleEscape);
+  }, [onClose]);
 
-    return (
-        <>
-            {(modal === "true" || modal === null) &&
-                <dialog className="fixed top-0 left-0 w-full h-full bg-transparent flex justify-center items-center z-50">
-                    <div className="relative w-full h-full flex justify-center items-center bg-black bg-opacity-20">
-                        <div className="bg-white p-5 rounded-lg shadow-lg max-w-lg w-full text-center animate-slide-up">
-                            <div className="text-lg text-[#bd9527] leading-6 font-myriad">
-                                <p>Coming Soon: NUANSA 2025</p>
-                                <br />
-                                <Link href="?modal=false">
-                                    <button type="button" className="bg-[#fecd94] text-white border-none py-2 px-4 rounded cursor-pointer text-base hover:bg-[#0056b3]">
-                                        Continue
-                                    </button>
-                                </Link>
-                            </div>
-                        </div>
-                    </div>
-                </dialog>
-            }
-        </>
-    );
+  return (
+    <div
+      id="modal-overlay"
+      className="fixed inset-0 bg-black bg-opacity-50 flex justify-center items-center z-50"
+      onClick={handleOutsideClick}
+    >
```

Comment:
Viewing on mobile, the box width exceeds the window width. For now, we should make it so that left and right of the modal has enough space, and add property `overflow-scroll`.